drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/exporters/hmis_export_fetcher.rb

```diff
@@ -0,0 +1,42 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# The export is generated elsewhere. This just orchestrates running the job and
+# returning the result.
+
+module HmisExternalApis::AcHmis::Exporters
+  class HmisExportFetcher
+    include Rails.application.routes.url_helpers
+
+    attr_accessor :export
+
+    delegate :content, to: :export
+
+    def run!
+      data_source = HmisExternalApis::AcHmis.data_source
+      user = User.system_user
+      version = '2024'
+
+      filter = ::Filters::HmisExport.new(data_source_ids: [data_source.id], version: version, user_id: user.id)
+
+      Rails.logger.info 'Generating HMIS CSV Export'
+
+      # FIXME: There's got to be a better way than these two ways.
+      #
+      # job_info = filter.execute_job(report_url: warehouse_reports_hmis_exports_url)
+      # export_id = job_info.arguments[3][:args][1]
+      # self.export = ::GrdaWarehouse::HmisExport.find(export_id)
+
+      filter.execute_job(report_url: warehouse_reports_hmis_exports_url)
+
+      self.export = ::GrdaWarehouse::HmisExport
+        .where(user_id: user.id)
+        .where(version: version)
+        .order('created_at desc')
+        .first!
```

Comment:
I think I prefer the commented-out version.. .at least then we don't run the risk of uploading the _wrong_ file to their server. @eanders wdyt?

Reply:
Oh, I missed that distinction, yeah, we definitely want to avoid uploading the wrong file.  I wish we had a better connection between the job and the export, but I wasn't able to find one.

Reply:
I'll switch it back. I had the first way originally. Mostly was fishing for something I was missing in how to do it "right"